src/ngx_http_lua_balancer.c

```diff
@@ -459,7 +459,7 @@ ngx_http_lua_balancer_save_session(ngx_peer_connection_t *pc, void *data)
 
 int
 ngx_http_lua_ffi_balancer_set_current_peer(ngx_http_request_t *r,
-    const u_char *addr, size_t addr_len, int port, char **err)
+    const u_char *addr, size_t addr_len, int port, int ssl, char **err)
```

Comment:
The semantics of this `ssl` parameter is very weird (0: off, 1: on, other values: default). Maybe we should introduce a separate C API function for setting SSL? Also, we may also want to dynamically configure whether to validate backend's SSL certificate and other things at the same time.